aten/src/ATen/native/RNN.cpp

```diff
@@ -195,6 +195,52 @@ struct QuantizedCellParamsDynamic {
   }
 };
 
+struct QuantizedCellParamsStatic {
+  QuantizedCellParamsStatic(const Tensor& _packed_w_ih,
+                            const Tensor& _packed_w_hh)
+    : packed_w_ih(_packed_w_ih), packed_w_hh(_packed_w_hh) {};
+
+  // use linear_prepack to get the packed weights.
+  const Tensor& packed_w_ih;
+  const Tensor& packed_w_hh;
+
+  // Assume [-8, 8] range
+  static constexpr float kQ8BitScaleWithRange16 = 0.0625;
+  static constexpr int kQ8BitZeroPoint = 0;
+
+  Tensor matmul_ih(Tensor input) const {
+    TORCH_CHECK(false, "matmul is not supported with quantized cell params");
+  }
+  Tensor matmul_hh(Tensor h) const {
+    TORCH_CHECK(false, "matmul is not supported with quantized cell params");
+  }
+  Tensor linear_ih(Tensor input) const {
+    const auto kFuncName = "quantized::linear";
+    const auto kOvrldName = "";
+    const std::vector<c10::IValue> output_ih_list =
+        callOp(kFuncName, kOvrldName, input, packed_w_ih,
+               kQ8BitScaleWithRange16, kQ8BitZeroPoint);
+    TORCH_INTERNAL_ASSERT(
+        output_ih_list.size() == 1,
+        "The output vector should have exactly one element");
+    const Tensor output_ih = output_ih_list[0].toTensor();
```

Comment:
Not 100% sure but I think this `const` might be preventing an RVO

Reply:
Yeah pretty sure that's the case: https://www.bfilipek.com/2017/01/const-move-and-rvo.html